jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/SpatialQuadTree.java

```diff
@@ -524,41 +514,41 @@ public void setBounds(Rectangle2D bounds) {
    * @param node
    */
   @Override
-  public void update(N node, Point2D location) {
-    //    LayoutModel<N, Point2D> nlayoutModel = (LayoutModel<N, Point2D>) layoutModel;
-    gridCache = null;
-    //    Point2D location = nlayoutModel.apply(node);
-    if (!this.getLayoutArea().contains(location)) {
-      log.trace(location + " outside of spatial " + this.getLayoutArea());
-      this.setBounds(this.getUnion(this.getLayoutArea(), location));
-      this.recalculate(layoutModel.getGraph().nodes());
-    }
-    Spatial locationContainingLeaf = getContainingQuadTreeLeaf(location);
-    log.trace("leaf {} contains {}", locationContainingLeaf, location);
-    TreeNode nodeContainingLeaf = getContainingQuadTreeLeaf(node);
-    log.trace("leaf {} contains node {}", nodeContainingLeaf, node);
-    if (locationContainingLeaf == null) {
-      log.trace("got null for leaf containing {}", location);
-    }
-    if (nodeContainingLeaf == null) {
-      log.trace("got null for leaf containing {}", node);
-    }
-    if (locationContainingLeaf != null && !locationContainingLeaf.equals(nodeContainingLeaf)) {
-      log.trace("time to recalculate");
-      this.recalculate(layoutModel.getGraph().nodes());
+  public void update(N node, Point location) {
+    if (isActive()) {
```

Comment:
again, I'd make this "if (!isActive()) return;" (early return) to make it slightly easier to grasp